src/InteractionModel.H

```diff
@@ -41,80 +42,178 @@ namespace ExaEpi
+      + amrex::DenseBins::build() creates the bin-sorted array of particle indices and
+        the offset array for each bin (where the offset of a bin is its starting location
+        in the bin-sorted array of particle indices).
+
+    + For each agent *i* in the bin-sorted array of agents:
+      + Find its bin and the range of indices in the bin-sorted array for agents in its bin
+      + If the agent is #Status::immune, do nothing.
+      + If the agent is #Status::infected with the number of days infected (RealIdxDisease::disease_counter)
+        less than the incubation length, do nothing.
+      + Else, for each agent *j* in the same bin:
+        + If the agent is #Status::immune, do nothing.
+        + If the agent is #Status::infected with the number of days infected (RealIdxDisease::disease_counter)
+          less than the incubation length, do nothing.
+        + Else if *i* is not infected and *j* is infected, compute probability of *i* getting infected
+          from *j* (see below).
+
+    Summary of how the probability of agent A getting infected from agent B is computed:
+    + Compute infection probability reduction factor from vaccine efficacy (#DiseaseParm::vac_eff)
+    + Within family - if their IntIdx::nborhood and IntIdx::family indices are same,
+      and the agents are at home:
+      + If B is a child, use the appropriate transmission probability (#DiseaseParm::xmit_child_SC or
+        #DiseaseParm::xmit_child) depending on whether B goes to school or not (#IntIdx::school)
+      + If B is an adult, use the appropriate transmission probability (#DiseaseParm::xmit_adult_SC or
+        #DiseaseParm::xmit_adult) depending on whether B works at a school or not (#IntIdx::school)
+*/
+template <typename AgentContainer, typename PTDType, typename CandidateFunc, typename BinaryInteractionFunc>
+void interactAgentsImpl(AgentContainer& agents, int bin_idx)
+{
+    BL_PROFILE("interactAgentsimpl");
+    int n_disease = agents.numDiseases();
+    IntVect bin_size = {AMREX_D_DECL(1, 1, 1)};
+    CandidateFunc isCandidate;
+    BinaryInteractionFunc binaryInteraction;
 
-                    AMREX_ALWAYS_ASSERT(np == bins_ptr->numItems());
+    for (int lev = 0; lev < agents.numLevels(); ++lev)
+    {
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+        for (MFIter mfi = agents.MakeMFIter(lev, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+            auto& ptile = agents.ParticlesAt(lev, mfi);
+            const auto& ptd = ptile.getParticleTileData();
+            const auto np = ptile.GetArrayOfStructs().numParticles();
+            auto& soa = ptile.GetStructOfArrays();
+            auto num_tiles = numTilesInBox(mfi.validbox(), true, bin_size);
+
+            int max_group = 0;
+            if (bin_idx != -1) {
+                auto bingroup_ptr = soa.GetIntData(bin_idx).data();
+                for (int i = 0; i < np; i++) {
```

Comment:
This looks like it relies on managed memory to pull the data off the device and run on the host. This could instead use `amrex::ParticleReduce` as here: https://github.com/AMReX-Codes/ExaEpi/blob/development/src/AgentContainer.cpp#L576 to do this calculation on the GPU. 

To turn off managed memory, you can do `amrex.the_arena_is_managed=0`

Reply:
Turning off managed memory causes a segfault at `read_workerflow` (line 552) at the call to 
`amrex::The_Arena()->alloc`. Am I correct in thinking that we require managed memory to use this call?

Reply:
Yes, it looks like it does. I can make a fix for that quickly.

Reply:
This branch now works fine with managed memory turned off, as long as the `agent.fast=true` parameter is set (that forces use of the GPU binning policy). I see that in main.cpp, line 28, the default is set to managed memory. Can we set this only if not using the fast parameter?